neuro_san/session/direct_concierge_session.py

```diff
@@ -51,10 +51,10 @@ def list(self, request_dict: Dict[str, Any]) -> Dict[str, Any]:
                     protobuf structure. Has the following keys:
                 "agents" - the sequence of dictionaries describing available agents
         """
-        manifest_restorer = RegistryManifestRestorer()
-        manifest_registries: Dict[str, AgentToolRegistry] = manifest_restorer.restore()
-
-        agents_list = []
-        for agent_name, _ in manifest_registries.items():
+        tool_factory_provider: ServiceToolFactoryProvider = \
+            ServiceToolFactoryProvider.get_instance()
+        agents_names: List[str] = tool_factory_provider.get_agent_names()
+        agents_list: List[Dict[str, Any]] = []
+        for agent_name in agents_names:
             agents_list.append({"agent_name": agent_name, "description": ""})
```

Comment:
For list of currently available agents, go to the "source of truth" - ServiceToolFactoryProvider own state.